asyncpg/compat.py

```diff
@@ -18,3 +22,23 @@ def aiter_compat(func):
 else:
     def aiter_compat(func):
         return func
+
+
+if PY_36:
+    def fspath(path):
+        if isinstance(path, os.PathLike):
+            path = os.fspath(path)
+        elif not isinstance(path, (str, bytes)):
+            path = None
```

Comment:
Maybe it's better to raise `TypeError` and catch it in the code that uses this function?